lib/grandpa/grandpa_test.go

```diff
@@ -966,6 +966,14 @@ func TestDeterminePreVote_NoPrimaryPreVote(t *testing.T) {
 
 	header, err := st.Block.BestBlockHeader()
 	require.NoError(t, err)
+
+	// the prevoted block should be 2 blocks behind the best block header
+	header, err = st.Block.GetHeader(header.ParentHash)
+	require.NoError(t, err)
+
+	header, err = st.Block.GetHeader(header.ParentHash)
+	require.NoError(t, err)
```

Comment:
Why is this twice here?

Also do we cover the case where the block is 3 blocks behind?